hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/impl/HddsDispatcher.java

```diff
@@ -329,6 +336,21 @@ private ContainerCommandResponseProto dispatchRequest(
     }
   }
 
+  private void updateBCSID(Container container,
+      DispatcherContext dispatcherContext, ContainerProtos.Type cmdType) {
+    long bcsID = container.getBlockCommitSequenceId();
+    long containerId = container.getContainerData().getContainerID();
+    Map<Long, Long> container2BCSIDMap;
+    if (dispatcherContext != null && (cmdType == ContainerProtos.Type.PutBlock
+        || cmdType == ContainerProtos.Type.PutSmallFile)) {
+      container2BCSIDMap = dispatcherContext.getContainer2BCSIDMap();
+      Preconditions.checkNotNull(container2BCSIDMap);
+      Preconditions.checkArgument(container2BCSIDMap.containsKey(containerId));
+      // updates the latest BCSID on every putBlock or putSmallFile
+      // transaction over Ratis.
+      container2BCSIDMap.computeIfPresent(containerId, (u, v) -> v = bcsID);
```

Comment:
`computeIfPresent` is not needed here, can be replaced with `Map#put`.

Reply:
will address in the next patch.